decisions/adr-ucum-implementation.md

```diff
@@ -0,0 +1,139 @@
+
+- how existing NMDC (meta)data was migrated and restructured into a UCUM‑valid format,  
+- how annotations are used to require UCUM units, slot requirements, and how new slots added to the NMDC schema must comply,  
+- the ETL updates required for NMDC's data ingest (work in progress),  
+- how ingested data can be checked and validated,  
+- updates to the submission portal ingest and validation.
+
+## Considered Options
+
+- Create various subclasses of `QuantityValue` for slots we want constrained, example `QuantityValueTemp` where `has_unit` is required and constrained to 'C'. Downside to approach was that a separate `QuantityValue` subclass had to be made for each `has_unit` value we wanted to require.
+- Create subclass `QuantityValuewithUnit` where `has_unit` is a required slot and constrained to enum `UnitEnum`. Downside to approach was that any of the `UnitEnum` values could be used for any slot (i.e. 'm' could be used for `temp` slot)
+
+- **Numeric‑only slots with LinkML `unit` metaslot** – Convert all `QuantityValue` slots to scalar numeric types (int/float) and use the LinkML `unit` metaslot to store UCUM symbols, which can represent one or multiple allowed units per slot. This simplifies validation but requires a major schema redesign and breaks compatibility with existing data.  
+- **Conversion pipeline on ingest** – Accept any unit in the submission schema and perform extensive unit conversion during ETL ingestion. Flexible for submitters but adds significant engineering overhead and runtime conversion complexity.  
+- **Hybrid approach (Chris Mungall’s suggestion)** – Keep the existing `QuantityValue` structure, constrain allowed units to the UCUM symbol for the MIxS‑preferred unit(s), and auto‑convert full‑name units to UCUM on ingest. Balances MIxS alignment with practical data handling.  
+- **Exploratory `range_expression` association** – Instead of using `storage_units` annotations, associate slots and units via a `range_expression` (as discussed in PR 2599). This would be an experimental LinkML feature applicable to any project, not just the UCUM case.  
+- **Chosen solution** – Introduce a `storage_units` annotation that lists allowed UCUM symbols per slot and enforce it via validation tests. This aligns with the hybrid approach while providing a clear, maintainable mechanism for unit standardisation.
+
+## Decision Outcomes
+
+Upon implementation of UCUM, all NMDC slots with `range: QuantityValue` will have a `storage_units` annotation to designate allowed units and required formats for data.
+
+## Validation
+
+The UCUM implementation introduces several layers of validation:
+
+- **Intrinsic schema validation** – LinkML’s built‑in validation checks that each `QuantityValue` slot either has a non‑empty `storage_units` annotation or a `units_alignment_excuse`. This is enforced by the autogenerated schema tests (`tests/test_units_alignment.py`).  
```

Comment:
When I read the phase "LinkML's built-in validation" I think it's referring to https://linkml.io/linkml/data/validating-data.html. It's unclear to me what that has to do with what is described in the rest of the sentence. Maybe it should say something like "We wrote an automated test that checks that each `QuantityValue`...". There is also nothing "autogenerated" about that test?